### PR TITLE
Split out the dependencies parsing and merging from go() 

### PR DESCRIPTION
**Pull request by andresdelfino**
This addresses #149.

### CODE REVIEW

**Comment by andresdelfino** [comment]
Don't know why fades/main.py is reported as a conflicting file. My branch is up to date with the tip of master.

File: fades/main.py [hunk 1]
```diff
@@ -56,13 +57,48 @@
...
 
 
-def _merge_deps(deps):
-    """Merge all the dependencies; latest dicts overwrite first ones."""
-    final = {}
-    for dep in deps:
+def consolidate_dependencies(needs_ipython, child_program,
+                             requirement_files, manual_dependencies):
+    """Parse files, get deps and merge them."""
+    # Deps specified first are overwritten by those specified last
```
Comment (facundobatista) [inline fades/main.py]: This comment is about the generic behaviour of the whole function. It should be part of the docstring.

File: tests/__init__.py [hunk 2]
```diff
@@ -34,3 +36,14 @@ def clean():
     testcase.addCleanup(clean)
 
     return tempfile
+
+
+@nottest
+def generate_test_file(testcase, lines):
+    tempfile = get_tempfile(testcase)
+
+    with open(tempfile, 'w') as f:
```
Comment (facundobatista) [inline tests/__init__.py]: Don't leave `encoding` unspecified, as it's platform dependent and we don't want that.

File: tests/__init__.py [hunk 3]
```diff
@@ -34,3 +36,14 @@ def clean():
     testcase.addCleanup(clean)
 
     return tempfile
+
+
+@nottest
+def generate_test_file(testcase, lines):
```
Comment (facundobatista) [inline tests/__init__.py]: Let's keep the previous naming style and call this a "temp" file. Maybe calling the function `create_tempfile`? A nice side effect is that you will not need the ugly `@nottest`

File: tests/test_main.py [hunk 4]
```diff
@@ -41,55 +44,118 @@ def test_prefix_is_not_baseprefix(self):
...
+    """Tests for the gathering stage of consolidate_dependencies."""
+
+    def test_needs_ipython(self):
+        d = main.consolidate_dependencies(needs_ipython=True, child_program=None,
+                                          requirement_files=None, manual_dependencies=None)
+
+        self.assertDictEqual(d, {'pypi': {Requirement.parse('ipython')}})
+
+    def test_child_program(self):
+        child_program = generate_test_file(self, ['"""fades:', 'dep', '"""'])
```
Comment (facundobatista) [inline tests/test_main.py]: Probably it's easier/cleaner to just use `tests/test_files/req_module.py` here?

File: fades/main.py (hunk 1 above)
Comment (andresdelfino) [reply fades/main.py]: Yeah, I had some trouble trying to make it all fit in a single < 100 chars line :sweat_smile: Changed it.

File: tests/__init__.py (hunk 3 above)
Comment (andresdelfino) [reply tests/__init__.py]: create_tempfile sounds good to me.

File: tests/test_main.py (hunk 4 above)
Comment (andresdelfino) [reply tests/test_main.py]: Yeap, didn't look into those test files.

**Comment by andresdelfino** [comment]
@facundobatista requested changes addressed.

**Comment by facundobatista** [comment]
Awesome! Thank you very much!

**Pull request merged**